multicluster/controllers/multicluster/member/serviceexport_controller.go

```diff
@@ -281,11 +289,16 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			eps.Subsets = []corev1.EndpointSubset{svcIPAsSubset}
 		}
```

Comment:
For the ClusterIP endpointIPType, `epNoChange` is always false? If that is indeed the case, please add some comments to explain that, as it is a little hard to understand.

Reply:
`epNoChange` should be always true considering the ClusterIP of a Service won't change in most cases. I refined a log which might help to understand it.
```
klog.InfoS("Service's Endpoints (PodIP or ClusterIP) has been converted into ResourceExport and no change, skip it", "Service",
```

Reply:
Could you explain what does svcNoChange and epNoChange mean? If svcNoChange means no change in the exported Service's spec, it does not apply to `EndpointIPTypePodIP`? If epNoChange  means no change in the exported Service's Endpoints, it does not apply to `EndpointIPTypeClusterIP`?

If it is true that EndpointIPTypePodIP just cares about Endpoints change, and EndpointIPTypeClusterIP just cares about Service spec change (?), we may just use one shared boolean to indicate whether we need to update the ResourceExport?

Reply:
The reason is we will create two ResourceExports for one ServiceExport. One is Service type of ResourceExport, another one is Endpoints type. No matter it's EndpointIPTypePodIP or EndpointIPTypeClusterIP, we will construct an Endpoint type of ResourceExport, the ResourceExport includes all available Endpoints (PodIP or ClusterIP). `epNoChange` is to track if all available Endpoints (PodIP or ClusterIP) are changed or not.
`svcNoChange` is to track if Service's ports and ClusterIP are changed or not. It's for Service type of ResourceExport, not for Endpoints type of ResourceExport.

Reply:
I asked in an earlier thread too - do you mean we include Service ClusterIP and Ports in the ResourceExport for a Service? Why? If we do not export ClusterIP and Ports in a Service ResourceExport, then seems EndpointIPTypePodIP needs not to know about ClusterIP? And in any case, EndpointIPTypeClusterIP needs not to know about Endpoints (except there is any ready endpoint or not)?

Reply:
I think svcNoChange / epNoChange can mean two things:

1. Whether there is a change in the exported Service/Endpoints
2. Whether there should be a change in the Service/Endpoints ResourceExport
(if this case, I would rename the two booleans to something like updateSvcExport/updateEpxport)

You meant which of these two? In either case, could you revise the code to make it clearer? For example, if it is the first case, we should set epNoChange to false for `EndpointIPTypeClusterIP`, as we should not care about Endpoints change for that; if it is the second case, maybe add a comment to say epNoChange should be true, if there is a ClusterIP/Port change for `EndpointIPTypeClusterIP`.

Reply:
It's for the second case, I renamed the variable skipUpdateSvcResourceExport and skipUpdateEPResourceExport and add a comment for EndpointIPTypeClusterIP type.